### PR TITLE
The k'ois nerf to end all k'ois nerfs

### PR DESCRIPTION
**Pull request by BurgerLUA**
# Overview
K'ois bars are now replaced with a less dangerous variant with the same name. These K'ois bars do not cause medical nightmares to occur, however the dangerous ones are moved to the "contraband" section of the vending machine and can be vended for free.

## New Chemicals
Unfiltered K'ois can still be found in K'ois spores. They can be filtered by grinding them up and adding Cardox . Cooking recipes are untouched, and still contain unfiltered K'ois.

Platinum can now be put into reagent form by grinding up platinum.

Cardox is a new chemical designed by nanotrasen that eliminates liquid phoron. It can be made by mixing Platinum, Sterilzine, and Carbon.

### CODE REVIEW

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm [hunk 1]
```diff
@@ -1,19 +1,41 @@
...
+		M.nutrition += nutriment_factor * removed // For hunger and fatness
+		M.add_chemical_effect(CE_BLOODRESTORE, 6 * removed)
+	else
+		M.adjustToxLoss(1 * removed)
+		return
+	..()
+
 /datum/reagent/kois
-	name = "K'ois"
+	name = "Unfilited K'ois"
```
Comment (Lohikar) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm]: Spelling error.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm [hunk 2]
```diff
@@ -35,7 +57,7 @@
 	..()
 
 /datum/reagent/blackkois
-	name = "Modified K'ois"
+	name = "Pure K'ois"
```
Comment (Lohikar) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm]: Did bygone approve this? This kinda changes the connotation of this.

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 3]
```diff
@@ -550,6 +550,13 @@
 	required_reagents = list("mindbreaker" = 1, "acetone" = 1, "inaprovaline" = 1)
 	result_amount = 3
 
+/datum/chemical_reaction/koispasteclean
+	name = "Filtered K'ois"
+	id = "koispasteclean"
+	result = "koispasteclean"
+	required_reagents = list("koispaste" = 1, "water" = 1, "ammonia" = 1)
+	result_amount = 3
```
Comment (Lohikar) [inline code/modules/reagents/Chemistry-Recipes.dm]: so the water and ammonia become part of the k'ois?

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 4]
```diff
@@ -431,10 +431,23 @@
 
 /obj/item/weapon/reagent_containers/food/snacks/koisbar/Initialize()
```
Comment (Lohikar) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: This is not the 'clean' path.

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 5]
```diff
@@ -431,10 +431,23 @@
 
 /obj/item/weapon/reagent_containers/food/snacks/koisbar/Initialize()
 	. = ..()
-	reagents.add_reagent("koispaste", 10)
+	reagents.add_reagent("koispasteclean", 10)
 	reagents.add_reagent("phoron", 15)
 	bitesize = 5
 
+/obj/item/weapon/reagent_containers/food/snacks/koisbar_clean
```
Comment (Lohikar) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: This looks like it's not supposed to be the clean path.

**Comment by BurgerLUA** [comment]
Possible todo: Make it so that only K'ois can spore if the plant contains unfiltered K'ois, making spraying ammonia and water everywhere a viable solution to a k'ois outbreak. Also add the ability to filter filtered k'ois back into regular k'ois by adding certain ingredients to remove the ammonia and water.

File: code/modules/reagents/Chemistry-Recipes.dm (hunk 3 above)
Comment (BurgerLUA) [reply code/modules/reagents/Chemistry-Recipes.dm]: yes, it would be like mixing a new drink

Comment (Lohikar) [reply code/modules/reagents/Chemistry-Recipes.dm]: That doesn't make sense if they're supposed to be 'cleaning' it.

**Comment by Alberyk** [comment]
You could just make so that the infectio is tied to a var or something, so you do not need to copy and paste most of the k'ois code.

**Comment by Bygonehero** [comment]
If the filtering process was changed to use Cardox I would be fine with the PR, otherwise, its implementation would clash with lore already established.

**Comment by BurgerLUA** [comment]
Accidentally thought antmen were vox for a secod, replaced IS_VOX with IS_VAURCA

File: code/game/machinery/vending_types.dm [hunk 6]
```diff
@@ -139,10 +139,11 @@
 		/obj/item/weapon/reagent_containers/food/snacks/meatsnack = 2,
 		/obj/item/weapon/reagent_containers/food/snacks/maps = 2,
 		/obj/item/weapon/reagent_containers/food/snacks/nathisnack = 2,
-		/obj/item/weapon/reagent_containers/food/snacks/koisbar = 4
+		/obj/item/weapon/reagent_containers/food/snacks/koisbar_clean = 4
 	)
 	contraband = list(
-		/obj/item/weapon/reagent_containers/food/snacks/syndicake = 6
+		/obj/item/weapon/reagent_containers/food/snacks/syndicake = 6,
+		/obj/item/weapon/reagent_containers/food/snacks/koisbar = 4
```
Comment (LordFowl) [inline code/game/machinery/vending_types.dm]: I don't really see any reason to keep these.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm [hunk 7]
```diff
@@ -1,19 +1,41 @@
 /* Food */
+/datum/reagent/kois_clean
```
Comment (LordFowl) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm]: Why must you hurt me this way? Make this reagent a child of K'ois so you're not literally copy-pasting _all_ of K'ois code. Then either override the affect_x() procs or do as Alberyk suggests and make it dependent on some variable defined in master K'ois code.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm [hunk 8]
```diff
@@ -1,19 +1,41 @@
...
 	var/injectable = 0
 	color = "#dcd9cd"
 	taste_description = "boiled cabbage"
 	unaffected_species = IS_MACHINE
 
 /datum/reagent/kois/affect_ingest(var/mob/living/carbon/M, var/alien, var/removed)
 	if(isvaurca(M))
-		M.heal_organ_damage(1.2 * removed, 1.2 * removed)
-		M.adjustToxLoss(-1.2 * removed)
+		M.heal_organ_damage(2.4 * removed, 2.4 * removed)
```
Comment (LordFowl) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm]: Ah, the classic tactic of sneaking in ridiculous buffs along with a moderate nerf. Any particular reason we need to double K'ois' healing efficacy? If you want to make the two distinct, then nerf the clean K'ois'. 2.4*removed is simply ridiculous, especially when unfiltered K'ois is the _easy_ reagent to procure.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm [hunk 9]
```diff
@@ -1,19 +1,41 @@
...
+	..()
+
 /datum/reagent/kois
-	name = "K'ois"
+	name = "Unfiltered K'ois"
 	id = "koispaste"
 	description = "A thick goopy substance, rich in K'ois nutrients."
 	metabolism = REM * 4
-	var/nutriment_factor = 10
+	var/nutriment_factor = 20
```
Comment (LordFowl) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm]: See below.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm [hunk 10]
```diff
@@ -203,6 +203,14 @@
 				new /obj/effect/decal/cleanable/greenglow(T)
 			return
 
+/datum/reagent/platinum
+	name ="Platinum"
+	id = "platinum"
+	description = "Platinum is a naturally occuring silvery metalic element."
+	reagent_state = SOLID
+	color = "#E0E0E0"
+	taste_description = "a watch"
```
Comment (LordFowl) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm]: Metallic watches are typically made of steel, I find this description offensive.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm [hunk 11]
```diff
@@ -360,6 +368,30 @@
 	if(volume >= 1)
 		T.wet_floor(2)
 
+/datum/reagent/cardox
```
Comment (LordFowl) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm]: Get in touch with BygoneHero and possibly Skull132 regarding cardox - from what I can recall there was discussion and possibly work between those two regarding a more comprehensive and lore-friendly implementation.

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 12]
```diff
@@ -550,6 +550,28 @@
...
 	result_amount = 3
 
+/datum/chemical_reaction/cardox
+	name = "Cardox"
+	id = "cardox"
+	result = "cardox"
+	required_reagents = list("platinum" = 1, "carbon" = 1, "sterilzine" = 1)
+	result_amount = 3
+
+/datum/chemical_reaction/cardox_removal
```
Comment (LordFowl) [inline code/modules/reagents/Chemistry-Recipes.dm]: See Lohikar's comments about the ammonia - cardox can't be a cleaner if it's absorbed in the reaction, it needs to be a catalyst.

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 13]
```diff
@@ -422,19 +422,32 @@
...
 
-/obj/item/weapon/reagent_containers/food/snacks/koisbar/Initialize()
+/obj/item/weapon/reagent_containers/food/snacks/koisbar_clean/Initialize()
 	. = ..()
-	reagents.add_reagent("koispaste", 10)
+	reagents.add_reagent("koispasteclean", 10)
 	reagents.add_reagent("phoron", 15)
 	bitesize = 5
 
+/obj/item/weapon/reagent_containers/food/snacks/koisbar
```
Comment (LordFowl) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: There is no reason to keep this around.

File: html/changelogs/burgerbb-koisnerf.yml [hunk 14]
```diff
@@ -0,0 +1,8 @@
+
+author: BurgerBB
+
+delete-after: True
+
+changes:
+  - rscadd: "Added new reagant, filtered Kois which can be made by combining ammonia, water, and unfiltered Kios."
```
Comment (LordFowl) [inline html/changelogs/burgerbb-koisnerf.yml]: Update changelog to reflect cardox.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm [hunk 15]
```diff
@@ -360,6 +368,30 @@
...
+	name = "Cardox"
+	id = "cardox"
+	description = "Cardox is an expensive, NanoTrasen designed cleaner intended to eliminate liquid phoron stains from suits."
+	reagent_state = LIQUID
+	color = "#EEEEEE"
+	taste_description = "cherry"
+
+/datum/reagent/toxin/cardox/affect_touch(var/mob/living/carbon/M, var/alien, var/removed)
+	.. ()
+	if(alien == IS_VAURCA)
```
Comment (LordFowl) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm]: Use isvaurca(M) as is used everywhere else.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm [hunk 16]
```diff
@@ -360,6 +368,30 @@
...
+		affect_blood(M, alien, removed*0.25)
+
+/datum/reagent/cardox/affect_blood(var/mob/living/carbon/M, var/alien, var/removed)
+	if(alien == IS_VAURCA)
```
Comment (LordFowl) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm]: Use isvaurca(M).

File: code/game/machinery/vending_types.dm (hunk 6 above)
Comment (BurgerLUA) [reply code/game/machinery/vending_types.dm]: Possible antag options for those who seeked approval.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm (hunk 7 above)
Comment (BurgerLUA) [reply code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm]: I was following your code distinction of black k'ois

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm (hunk 8 above)
Comment (BurgerLUA) [reply code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm]: most antitoxin medicine heals 4-6 units, but I'll revert it if it's an issue

File: code/modules/reagents/Chemistry-Recipes.dm (hunk 12 above)
Comment (BurgerLUA) [reply code/modules/reagents/Chemistry-Recipes.dm]: I was under the impression that cleaners lose a bit of their strength when used, hence the 0.1 unit removal.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm (hunk 7 above)
Comment (LordFowl) [reply code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm]: Black K'ois arguably could be a child as well, but the distinction is that it's infection is quite different where as the clean k'ois' infection is literally just an if() check at most.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm (hunk 8 above)
Comment (LordFowl) [reply code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm]: Most anti-toxin also doesn't simultaneously heal brute and burn damage.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm (hunk 15 above)
Comment (Lohikar) [reply code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm]: `isvaurca()` checks should not be used in reagent code.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm [hunk 17]
```diff
@@ -360,6 +368,30 @@
...
+	name = "Cardox"
+	id = "cardox"
+	description = "Cardox is an expensive, NanoTrasen designed cleaner intended to eliminate liquid phoron stains from suits."
+	reagent_state = LIQUID
+	color = "#EEEEEE"
+	taste_description = "cherry"
+
+/datum/reagent/toxin/cardox/affect_touch(var/mob/living/carbon/M, var/alien, var/removed)
+	.. ()
+	if(isvaurca(M))
```
Comment (Lohikar) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm]: This should be `if (alien == IS_VAURCA)`.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Dispenser.dm [hunk 18]
```diff
@@ -125,7 +125,7 @@
 	return
 
 /datum/reagent/ethanol/affect_ingest(var/mob/living/carbon/M, var/alien, var/removed)
-	if(isvaurca(M))//Vaurca are damaged instead of getting nutrients, but they can still get drunk
+	if(alien && alien == IS_VAURCA)//Vaurca are damaged instead of getting nutrients, but they can still get drunk
```
Comment (LordFowl) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Dispenser.dm]: Shouldn't these be

> alien && alien =? isvaurca()

?

Comment (Lohikar) [reply code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Dispenser.dm]: That makes no sense, fowl.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm [hunk 19]
```diff
@@ -11,7 +11,7 @@
 	var/kois_type = 1
 
 /datum/reagent/kois/affect_ingest(var/mob/living/carbon/M, var/alien, var/removed)
-	if(isvaurca(M))
+	if(alien && alien == IS_VAURCA)
```
Comment (Lohikar) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm]: Why are you checking for null here?

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm [hunk 20]
```diff
@@ -92,10 +92,11 @@
 	affect_ingest(M, alien, removed)
 
 /datum/reagent/nutriment/affect_ingest(var/mob/living/carbon/M, var/alien, var/removed)
-	if(isvaurca(M))
-		M.adjustToxLoss(1.5 * removed)
-	if(alien && alien == IS_UNATHI)
-		return
+	if(alien)
```
Comment (Lohikar) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm]: Why not just make this entire thing a switch?
```dm
switch (alien)
    if (IS_VAURCA)
        M.adjustToxLoss(1.5 * removed)
    if (IS_UNATHI)
        return
    else
        digest(M, removed)
```

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm [hunk 21]
```diff
@@ -378,11 +378,11 @@
 
 /datum/reagent/toxin/cardox/affect_touch(var/mob/living/carbon/M, var/alien, var/removed)
 	.. ()
-	if(isvaurca(M))
+	if(alien == IS_VAURCA)
 		affect_blood(M, alien, removed * 0.25)
 
 /datum/reagent/cardox/affect_blood(var/mob/living/carbon/M, var/alien, var/removed)
-	if(isvaurca(M))
+	if(alien && alien == IS_VAURCA)
```
Comment (Lohikar) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm]: Why are you checking for null?

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm [hunk 22]
```diff
@@ -60,7 +60,7 @@
 /datum/reagent/toxin/phoron/affect_blood(var/mob/living/carbon/M, var/alien, var/removed)
 	if(ishuman(M))
 		var/mob/living/carbon/human/H = M
-		if(H.species.has_organ["filtration bit"] && isvaurca(H))
+		if(H.species.has_organ["filtration bit"] && alien && alien == IS_VAURCA)
```
Comment (Lohikar) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm]: Again, null check.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm [hunk 23]
```diff
@@ -85,7 +85,7 @@
 		L.adjust_fire_stacks(amount / 5)
 
 /datum/reagent/toxin/phoron/affect_touch(var/mob/living/carbon/M, var/alien, var/removed)
-	if(isvaurca(M))
+	if(alien && alien == IS_VAURCA)
```
Comment (Lohikar) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm]: More null checks.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm (hunk 20 above)
Comment (BurgerLUA) [reply code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm]: Vaurca can still digest nutrients.

Comment (BurgerLUA) [reply code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm]: oh nvm you're right

File: html/changelogs/burgerbb-koisnerf.yml [hunk 24]
```diff
@@ -0,0 +1,8 @@
+
+author: BurgerBB
+
+delete-after: True
+
+changes:
+  - rscadd: "Added filtered Kois, a significantly less dangerous Kois which can be made by combining Cardox and normal Kios. Added Cardox, an anti-phoron reagent which can eliminate phoron and remove the harmful spores from Kois."
```
Comment (skull132) [inline html/changelogs/burgerbb-koisnerf.yml]: Typo: kios vs kois.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm [hunk 25]
```diff
@@ -9,67 +8,48 @@
...
-					infest.replaced(H, affected)
-
-				else
-					H.adjustToxLoss(1 * removed)
-		else
-			M.adjustToxLoss(1 * removed)
-		return
+			switch(kois_type)
+				if(1) //Normal
+					if(!H.internal_organs_by_name["kois"] && prob(10*removed))
```
Comment (skull132) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm]: This probability, and the one below it should be reduced by some margin. After speaking with Fowl about this, we determined that kois infections should be "rare". Atm, this evaluates to `10 * 0.2 * 4 = 8%` every Life() tick, until you've managed to ingest all of it. This is an incredibly high chance from a statistical perspective, and could probably be slashed, if not brought down by an order of magnitude.

**Comment by skull132** [comment]
@LordFowl @Lohikar update your reviews as is appropriate.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm (hunk 25 above)
Comment (BurgerLUA) [reply code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm]: Is there a specific value you desire?

**Pull request merged**